storm-core/src/jvm/org/apache/storm/trident/Stream.java

```diff
@@ -565,19 +578,155 @@ public Stream aggregate(Fields inputFields, ReducerAggregator agg, Fields functi
                 .aggregate(inputFields, agg, functionFields)
                 .chainEnd();
     }
-    
+
+    /**
+     * Returns a stream of tuples which are aggregated results of a tumbling window with every {@code windowCount} of tuples.
+     *
+     * @param windowCount represents no of tuples in the window
+     * @param windowStoreFactory intermediary tuple store for storing windowing tuples
+     * @param inputFields projected fields for aggregator
+     * @param aggregator aggregator to run on the window of tuples to compute the result and emit to the stream.
+     * @param functionFields fields of values to emit with aggregation.
+     *
+     * @return
+     */
+    public Stream tumblingWindow(int windowCount, WindowsStoreFactory windowStoreFactory,
+                                      Fields inputFields, Aggregator aggregator, Fields functionFields) {
+        return window(TumblingCountWindow.of(windowCount), windowStoreFactory, inputFields, aggregator, functionFields);
+    }
+
+    /**
+     * Returns a stream of tuples which are aggregated results of a sliding window with every {@code windowCount} of tuples
+     * and slides the window with {@code slideCount}.
+     *
+     * @param windowCount represents tuples count of a window
+     * @param slideCount represents sliding count window
```

Comment:
Make it in sync with core api - "@param slidingInterval the number of tuples after which the window slides"